server/controllers/profile.js

```diff
@@ -1,39 +1,67 @@
 import User from '../models/userModels.js';
 import jwt from 'jsonwebtoken';
+import validator from 'validator';
 
 export const porfile_put = (profileType) => async (req, res) => {
-	const { name, publicEmail, isActive, description, techStack } = req.body;
-	const token = req.cookies.access_token;
+  const { name, publicEmail, isActive, description, techStack } = req.body;
+  const token = req.cookies.access_token;
 
-	try {
-		const decoded = jwt.verify(token, process.env.TOKEN_SECRET);
-		const updatedUser = await User.updateOne(
-			{ _id: decoded },
-			{
-				name,
-				publicEmail,
-				[profileType]: {
-					isActive,
-					description,
-					techStack,
-				},
-			}
-		).exec();
-		res.status(201).json(updatedUser);
-	} catch (error) {
-		res.status(409).json({ message: error.message });
-	}
+  try {
+    if (!validator.isEmail(publicEmail))
+      throw new Error('Please enter a valid email');
```

Comment:
We figured put the validator for `publicEmail` which is set in `server/models/userModels` was not working, so we added the code here `server/controllers/profile.js` to validate it.